x-pack/plugins/security_solution/scripts/endpoint/resolver_generator_script.ts

```diff
@@ -179,35 +220,70 @@ async function main() {
       type: 'boolean',
       default: false,
     },
+    newUser: {
+      alias: 'nu',
+      describe: 'Adds a new user with username:password',
+      type: 'string',
+      default: 'endpoint_user:changeme',
+    },
   }).argv;
   let ca: Buffer;
-  let kbnClient: KbnClient;
+
   let clientOptions: ClientOptions;
+  let url: string;
+  let node: string;
+  const toolingLogOptions = {
+    log: new ToolingLog({
+      level: 'info',
+      writeTo: process.stdout,
+    }),
+  };
+
+  let kbnClientOptions: KbnClientOptions = {
+    ...toolingLogOptions,
+    url: argv.kibana,
+  };
 
   if (argv.ssl) {
     ca = fs.readFileSync(CA_CERT_PATH);
-    const url = argv.kibana.replace('http:', 'https:');
-    const node = argv.node.replace('http:', 'https:');
-    kbnClient = new KbnClient({
-      log: new ToolingLog({
-        level: 'info',
-        writeTo: process.stdout,
-      }),
+    url = argv.kibana.replace('http:', 'https:');
+    node = argv.node.replace('http:', 'https:');
+    kbnClientOptions = {
+      ...kbnClientOptions,
       url,
       certificateAuthorities: [ca],
-    });
+    };
+
     clientOptions = { node, tls: { ca: [ca] } };
   } else {
-    kbnClient = new KbnClient({
-      log: new ToolingLog({
-        level: 'info',
-        writeTo: process.stdout,
-      }),
-      url: argv.kibana,
-    });
     clientOptions = { node: argv.node };
   }
-  const client = new Client(clientOptions);
+  let client = new Client(clientOptions);
+  // add endpoint user
+  const newUserCreds = argv.newUser ? argv.newUser.split(':') : undefined;
+  const user: UserInfo | undefined = await addUser(
```

Comment:
I'm a bit confused here. This means that if a user does not want to create a new user (`--newUser=""'`), then the tool would fail to run because I assume if the second argument below is `undefined`, the `addUser()` function will throw. I would propose that this code here only run if the `newUser` option is true.

The remainder of my comments will be following this comment (that `newUser` could be un-set or not used)

Reply:
No the `addUser()` doesn't throw if one doesn't provide values for the user `--newUser=""`. I went with the assumption to always add a user with the roles we need, so there're default values for the params.

Reply:
No the `addUser()` doesn't throw if one doesn't provide values for the user `--newUser=""`. I went with the assumption to always add a user with the roles we need, so there're default values for the params. Although now, I see that assumption is moot.

Reply:
done 5ce10629642bcdf4d08368f8b1f53374601d1046.